### PR TITLE
CrowdWorking: Type and dummy-platform

### PR DESCRIPTION
**Pull request by LeanderK**
- implemented `DummyPlatform` used for testing/exploring the api/verifying the worker-service is responsive
- implemented `FallbackWorker`
- added `getType` to platform
- added missing patch worker rest-command

### CODE REVIEW

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/Platform.java [hunk 1]
```diff
@@ -11,33 +11,42 @@
...
     String getName();
 
     /**
+     * Returns the type of the platform.
+     * <p>
+     * The String must match {@code [a-z0-9_]+} and must be a constant since it is the only way other services can
+     * communicate with the platform.
+     * @return a String
+     */
+    String getType();
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/Platform.java:37]: isnt there some fancy way to mark something "final for the class?"

Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/Platform.java:37]: no, unfortunately not

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/FallbackWorker.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package edu.kit.ipd.crowdcontrol.objectservice.crowdworking;
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/FallbackWorker.java]: Does that really belong in the crowdworking package ? I mean its the rest api which defines that its the field email

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/FallbackWorker.java]: Does that really belong in the crowdworking package ? I mean its the rest api which defines that its the field email

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/Platform.java (hunk 1 above)
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/Platform.java:37]: :(

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java [hunk 3]
```diff
@@ -52,12 +69,13 @@ public PlatformManager(List<Platform> crowdPlatforms, WorkerIdentification fallb
 
         //create hashmap of platforms
         platforms = crowdPlatforms.stream()
-                .collect(Collectors.toMap(Platform::getName, Function.identity()));
+                .collect(Collectors.toMap(Platform::getType, Function.identity()));
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java:55]: this breaks the manager! Assume two mturk platforms one productive one sandbox, the manager operations will now be executed on the same interface instance, ( I think the first which appears) but should not.

Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java:55]: these are the same type? not `mturk` and `murk_sandbox`?

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java:55]: Sure!

Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java:55]: but thats a problem! I need a non-changing id which can use to identify a platform. when i call `/workers/:platform/identity` i need a string `:platform` which i can use in the worker-service to call the platform. Then the name type is not right.

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java:55]: I never knew about that, add another map with the type as key, and use it in identify worker (we have to write somewhere that the idenfitifaction should not differ in 2 instances...

Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java:55]: It must be possible that the identification differs for two platforms of the same type. Could also be two totallly unrelated Pybossa instances.

**Pull request closed without merging**